packages/yarnpkg-fslib/sources/buffer.ts

```diff
@@ -0,0 +1,29 @@
+import {NativePath, PortablePath, Path, PathUtils, PathLike, PathType, ppath, npath} from './path';
+
+export class BasePathBuffer<P extends Path> extends Buffer {
+  declare public toString: (encoding?: string, start?: number, end?: number) => P;
```

Comment:
Rather than using `declare`, shouldn't `BasePathBuffer` be an abstract class?

Reply:
🤔 The `declare` is used here to augment the method in the superclass, since we can assure the compiler that `toString` will always return a path. It would be a good idea to turn these into abstract classes (even though they don't actually have any `abstract` fields / methods), but I don't see how that would change the way `toString` is currently typed - we only need to augment it, not to implement it and not to require subclasses to implement it.

Reply:
Thinking of this more, it might be a better idea to fully override the `toString` method to make it always return a path with the `latin1` encoding, since it's the only way we can make sure that a valid path will be returned.